test-clients/src/main/java/com/hedera/services/bdd/spec/verification/traceability/SidecarWatcher.java

```diff
@@ -49,12 +50,20 @@ public SidecarWatcher(final Path recordStreamFolderPath) {
     private final Queue<ExpectedSidecar> expectedSidecars = new LinkedBlockingDeque<>();
     private final Multimap<String, MismatchedSidecar> failedSidecars = HashMultimap.create();
 
-    private boolean shouldTerminateAfterNextSidecar = false;
+    private boolean noMoreIncomingExpectedSidecars = false;
     private boolean hasSeenFirst = false;
+    // watch service creates 2 notifications for each sidecar
+    // parsing the sidecar on the first notification throws an
+    // exception, since the file is not finalized at that point,
+    // so we need to know when the 2nd one is and only then parse
+    private boolean firstSidecarNotifyEventSeen = false;
```

Comment:
Q: Do subsequent side cars still get 2 notifications per file?
And so far as you start parsing on the 2nd notification of the very 1st side car then everything is good?

Reply:
> Do subsequent side cars still get 2 notifications per file?

Correct, each sidecar file gets 2 notification events in the watch service. If we log each event that is received, the logs will show that the main record stream file also gets two events (we compress it also), and the signature file gets only 1 (not compressed). If we run the node with compression disabled, as it was before we merged the compression PR, each file gets only 1 notification. So probably `GZIPOutputStream` creates the files in a slightly different way and that is why we now get two notifications. 

> And so far as you start parsing on the 2nd notification of the very 1st sidecar then everything is good?

Yes, but this applies to each sidecar file --- each sidecar file gets 2 notification events (not only the first sidecar file), and for each, we have to parse the file on the second event, or the final event we receive for the file. The first event probably is sent at an intermediary point in the file creation/directory modification and that is why if we try to parse the file then, we get an error. That's the whole purpose of the `firstSidecarNotifyEventSeen`, to keep track of the number of events received and parse only on the second one. A [StackOverflow post about the same issue ](https://stackoverflow.com/questions/16777869/java-7-watchservice-ignoring-multiple-occurrences-of-the-same-event)recommends sleeping the thread for 50ms, but to me personally, this solution seemed brittle and that is why I opted in for the flag.

Reply:
Resolving since now obsolete after removal of WatchServiceAPI.